.github/workflows/smoke.yml

```diff
@@ -128,6 +129,211 @@ jobs:
+    - name: Save Test Result
+      uses: actions/upload-artifact@v3
+      with:
+        name: benchmark-oci
+        path: misc/benchmark/${{env.IMAGE}}.csv
+
+  benchmark-nydus-no-prefetch:
+    runs-on: ubuntu-latest
+    needs: [contrib-build, nydus-build]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Download Nydus
+      uses: actions/download-artifact@master
+      with:
+        name: nydus-artifact
+        path: target/release
+    - name: Download Nydusify
+      uses: actions/download-artifact@master
+      with:
+        name: nydusify-artifact
+        path: contrib/nydusify/cmd
+    - name: Prepare Nydus Environment
+      run: |
+        sudo install -D -m 755 contrib/nydusify/cmd/nydusify /usr/local/bin
+        sudo install -D -m 755 target/release/nydusd target/release/nydus-image /usr/local/bin
+        wget https://github.com/containerd/nydus-snapshotter/releases/download/v0.7.3/nydus-snapshotter-v0.7.3-x86_64.tgz
+        tar zxvf nydus-snapshotter-v0.7.3-x86_64.tgz
+        sudo install -D -m 755 nydus-snapshotter/containerd-nydus-grpc /usr/local/bin/
+        sudo wget https://github.com/containerd/nerdctl/releases/download/v1.3.0/nerdctl-1.3.0-linux-amd64.tar.gz
+        sudo tar -xzvf nerdctl-1.3.0-linux-amd64.tar.gz -C /usr/local/bin
+        sudo mkdir -p /opt/cni/bin
+        sudo wget https://github.com/containernetworking/plugins/releases/download/v1.2.0/cni-plugins-linux-amd64-v1.2.0.tgz 
+        sudo tar -xzvf cni-plugins-linux-amd64-v1.2.0.tgz -C /opt/cni/bin
+        sudo install -D misc/benchmark/cni_bridge.conf /etc/cni/net.d/bridge.conf
+        sudo install -D misc/benchmark/nydusd_config.json /etc/nydus/config.json
+        sudo install -D misc/benchmark/containerd_config.toml /etc/containerd/config.toml
+        sudo systemctl restart containerd
+        sudo install -D misc/benchmark/nydus-snapshotter.service /etc/systemd/system/nydus-snapshotter.service
+        sudo systemctl start nydus-snapshotter
+    - name: Prepare Nydus Registry Environment
+      run: |
+        sudo docker run -d --restart=always -p 5000:5000 --name registry registry
+        sudo DOCKER_CONFIG=$HOME/.docker nydusify convert \
+          --source ${{env.IMAGE}}:latest \
+          --target localhost:5000/${{env.IMAGE}}:latest_nydus \
+          --fs-version 6 \
+          --platform linux/amd64,linux/arm64
+        git clone https://github.com/magnific0/wondershaper.git
+        sudo install -D -m 755 wondershaper/wondershaper /usr/local/bin
+    - name: BenchMark Test
+      run: |
+        cd misc/benchmark
+        sudo python3 benchmark.py --mode nydus-no-prefetch
+    - name: Save Test Result
+      uses: actions/upload-artifact@v3
+      with:
+        name: benchmark-nydus-no-prefetch
+        path: misc/benchmark/${{env.IMAGE}}.csv
+
+  benchmark-nydus-all-prefetch:
+    runs-on: ubuntu-latest
+    needs: [contrib-build, nydus-build]
+    steps:
```

Comment:
Can we reuse these steps `Download Nydus`, `Download Nydusify`, `Prepare Nydus Environment`, `Prepare Nydus Registry Environment`?

Reply:
Actually, I tried it but I failed. Is any example can I learn from.

Reply:
A possible approach is adding these as an individual job and upload these binary components and configs as an directory artifact, then download them on each benchmark job, but it doesn't work for `Prepare Nydus Registry Environment`.